packagedb/find_source_repo.py

```diff
@@ -0,0 +1,454 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# purldb is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/purldb for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import enum
+import logging
+import subprocess
+from typing import Generator
+from typing import List
+from urllib.parse import urlparse
+
+import requests
+from packageurl import PackageURL
+from packageurl.contrib.purl2url import get_download_url
+from packageurl.contrib.purl2url import purl2url
+from scancode.api import get_urls as get_urls_from_location
+
+from minecode.model_utils import add_package_to_scan_queue
+from minecode.visitors.maven import get_merged_ancestor_package_from_maven_package
+from packagedb.models import Package
+from packagedb.models import PackageContentType
+from packagedb.models import PackageSet
+
+logger = logging.getLogger(__name__)
+
+
+class URLDataReturnType(enum.Enum):
```

Comment:
Can you clarify what `url` or `text` means in this context? What exactly is being returned from the responses when you choose `url` or `text`?